File: examples/auth/magic-link.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Passwordless email link: request, then verify the token from the callback URL.
export async function magicLinkFlow(
  client: JsBaoClient,
  email: string,
  magicToken: string,
) {
  // #region example
  // Request the email. Requires oauthRedirectUri on the client, or pass
  // an explicit redirectUri here.
  await client.magicLinkRequest(email, {
    redirectUri: "https://app.example.com/auth/magic-callback",
  });

  // On the callback page, read ?magic_token=... and verify it.
  const { user, isNewUser, promptAddPasskey } =
    await client.magicLinkVerify(magicToken);
  // Token is now stored on the client and the WS auto-connects.
  // #endregion example
  return { user, isNewUser, promptAddPasskey };
}

File: examples/auth/magic-link.swift
import JsBaoClient

// Passwordless email link: request, then verify the token from the callback.
// On Swift `magicLinkRequest` takes an explicit `redirectUri`.
func magicLinkFlow(
  client: JsBaoClient,
  email: String,
  magicToken: String
) async throws {
  // #region example
  // Request the email. Pass the magic-link callback redirect URI.
  _ = try await client.auth.magicLinkRequest(
    email: email,
    redirectUri: "https://app.example.com/auth/magic-callback"
  )

  // On the callback, read the `magic_token` value and verify it.
  let result = try await client.auth.magicLinkVerify(token: magicToken)
  // Token is now stored on the client and the WS auto-connects.
  let user = result.user
  let isNewUser = result.isNewUser ?? false
  // #endregion example
  _ = (user, isNewUser)
}
